### PR TITLE
Add more tests for Playwright panel functionality

### PR DESCRIPTION
**Pull request by markerikson**
This PR:

- [x] Adds a Playwright version of `cypress-04` for testing test step list item interactions
- [x] Adds a Playwright version of `cypress-05` for testing DOM node interactions

### CODE REVIEW

File: packages/e2e-tests/tests/playwright-04_menu-commands.test.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+}) => {
+  await startTest(page, recordingId);
+  await openViewerTab(page);
+
+  await openPlaywrightTestPanel(page);
+
+  const testPanel = getTestSuitePanel(page);
+
+  const isVisible = await testPanel.isVisible();
+  expect(isVisible).toBe(true);
```
Comment (bvaughn) [inline packages/e2e-tests/tests/playwright-04_menu-commands.test.ts:32]: nit: `openPlaywrightTestPanel` helper already ensures the test suite panel is visible.

File: packages/e2e-tests/tests/playwright-05_hover-dom-previews.test.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+}) => {
+  await startTest(page, recordingId);
+  await openViewerTab(page);
+
+  await openPlaywrightTestPanel(page);
+
+  const testPanel = getTestSuitePanel(page);
+
+  const isVisible = await testPanel.isVisible();
+  expect(isVisible).toBe(true);
```
Comment (bvaughn) [inline packages/e2e-tests/tests/playwright-05_hover-dom-previews.test.ts:26]: nit: `openPlaywrightTestPanel` helper already ensures the test suite panel is visible.

File: packages/e2e-tests/tests/playwright-04_menu-commands.test.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  // Verify that's disabled.
+  await firstJ2CStep.click();
+  const menuButton = getByTestName(firstJ2CStep, "TestSectionRowMenuButton");
+  await menuButton.click();
+
+  const jumpToSourceMenuItem = page.getByText("Jump to test source");
+  expect(await jumpToSourceMenuItem.getAttribute("data-disabled")).toBe("true");
+
+  // click it anyway to hide the context menu
+  await jumpToSourceMenuItem.click({ force: true });
```
Comment (bvaughn) [inline packages/e2e-tests/tests/playwright-04_menu-commands.test.ts:85]: `page.keyboard.press("Escape")` is a less hacky way of doing this?

(or we could copy the `hideContextMenu` helper from our screenshot tests into the e2e-tests folder and use it, if you think that's cleaner)

File: packages/e2e-tests/tests/playwright-05_hover-dom-previews.test.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  const userActionSteps = steps.filter({
+    hasText: /click/,
+  });
+
+  // Hovering over a user action step should show a preview of the DOM node
+  const lastClickStep = userActionSteps.last();
+  await lastClickStep.scrollIntoViewIfNeeded();
+
+  // Note that the ID is dynamically generated based on the piece of the box model shown
+  const highlighter = page.locator("#box-model-content");
```
Comment (bvaughn) [inline packages/e2e-tests/tests/playwright-05_hover-dom-previews.test.ts:57]: What creates this `box-model-content` element? Searching our project for that string, I only see test selectors and a CSS selector.

File: packages/e2e-tests/tests/playwright-05_hover-dom-previews.test.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+  await waitFor(
+    async () => {
+      // Repeatedly hover over the first step and then the click step, to force the
+      // `onMouseEnter` handler to keep checking if we have a DOM node entry available.
```
Comment (bvaughn) [inline packages/e2e-tests/tests/playwright-05_hover-dom-previews.test.ts:62]: Is there a better way we could wait for this case? I think we typically use data attributes to signal readiness for cases like this (and then we read them, and `delay(...)` before reading again if necessary)

Comment (markerikson) [reply packages/e2e-tests/tests/playwright-05_hover-dom-previews.test.ts:62]: I don't think so. We don't use the DOM node cache at all while rendering. It's only checked in the `onHover` logic, synchronously, to see if we have it available.

Comment (bvaughn) [reply packages/e2e-tests/tests/playwright-05_hover-dom-previews.test.ts:57]: FWIW I think we could still do what I described here. The `useCacheStatus` hook (in "suspense") lets you peek at a cache status without kicking off a fetch.

**Pull request merged**